packages/dashboard/src/components/signup/signup.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { render, fireEvent, screen } from "@solidjs/testing-library";
+import { createSignal } from "solid-js";
+
+import SignUp from "./signup";
+
+// Mock external dependencies
+vi.mock("@solidjs/router", () => ({
+  useNavigate: vi.fn(),
+  useSearchParams: vi.fn(() => [
+    createSignal({
+      email: "test@example.com",
+      signup_id: "12345",
+      challenge: "abcde",
+    }),
+  ]),
+}));
+
+vi.mock("../utils", () => ({
+  arrayBufferToBase64: vi.fn(),
+}));
+
+global.fetch = vi.fn(() =>
```

Comment:
Consider using vi.stubGlobal instead of overwriting https://vitest.dev/guide/mocking.html#globals